pkg/yurtadm/util/kubernetes/kubernetes.go

```diff
@@ -122,29 +127,30 @@ func CheckAndInstallKubelet(kubernetesResourceServer, clusterVersion string) err
 	}
 
 	if !kubeletExist {
-		//download and install kubernetes-node
-		packageUrl := fmt.Sprintf(constants.KubeUrlFormat, kubernetesResourceServer, clusterVersion, runtime.GOARCH)
-		savePath := fmt.Sprintf("%s/kubernetes-node-linux-%s.tar.gz", constants.TmpDownloadDir, runtime.GOARCH)
+		//download and install kubelet
+		packageUrl := fmt.Sprintf(constants.KubeletUrlFormat, kubernetesResourceServer, clusterVersion, runtime.GOARCH)
+		savePath := fmt.Sprintf("%s/kubelet", constants.TmpDownloadDir)
 		if _, err := os.Stat(savePath); errors.Is(err, os.ErrNotExist) {
```

Comment:
when kubelet file can not be found by `exec.LookPath` but exist under `/tmp` dir, we should download the kubelet file and override the `/tmp/kubelet` file instead of using this `/tmp/kubelet` file. because we can not make sure the `/tmp/kubelet` is the correct kubelet.